.github/workflows/avh.yml

```diff
@@ -0,0 +1,96 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Arm Virtual Hardware
+
+on:
+    push:
+    pull_request:
+    workflow_dispatch:
+
+concurrency:
+    group: ${{ github.ref }}-${{ github.workflow }}-${{ (github.event_name == 'pull_request' && github.event.number) || (github.event_name == 'workflow_dispatch' && github.run_number) || github.sha }}
+    cancel-in-progress: false
+
+jobs:
+    arm_crosscompile:
+        name: Linux ARM Cross compile
+        timeout-minutes: 70
+
+        runs-on: ubuntu-latest
+        if: github.actor != 'restyled-io[bot]'
+
+        container:
+            image: ghcr.io/project-chip/chip-build-crosscompile:1
+            volumes:
+                - "/tmp/bloat_reports:/tmp/bloat_reports"
+
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v3
+
+            - name: Checkout submodules & Bootstrap
+              uses: ./.github/actions/checkout-submodules-and-bootstrap
+              with:
+                platform: linux
+
+            - name: Build Samples
+              timeout-minutes: 45
+              run: |
+                  ./scripts/run_in_build_env.sh \
+                     "./scripts/build/build_examples.py \
+                        --target linux-arm64-chip-tool-ipv6only-mbedtls-clang-minmdns-verbose \
+                        --target linux-arm64-light-ipv6only-mbedtls-clang-minmdns-verbose \
+                        build \
+                     "
+
+            - name: Upload built samples
+              uses: actions/upload-artifact@v3
+              with:
+                  name: arm_crosscompiled_samples
+                  path: |
+                      out/linux-arm64-chip-tool-ipv6only-mbedtls-clang-minmdns-verbose/chip-tool
+                      out/linux-arm64-light-ipv6only-mbedtls-clang-minmdns-verbose/chip-lighting-app
+
+    arm_e2e_tests:
+        name: Arm Virtual Hardware End to end tests
+        timeout-minutes: 10
+
+        runs-on: ubuntu-latest
+
+        env:
+            AVH_API_TOKEN: ${{ secrets.AVH_API_TOKEN }}
+            AVH_API_ENDPOINT: https://csa.app.avh.arm.com/api
+            AVH_PROJECT_NAME: "${{ github.workflow }} #${{ github.run_number }} - End to end tests"
+
+        needs: arm_crosscompile
+
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v3
+
+            - name: Downloads Cross-compiled samples
+              uses: actions/download-artifact@v3
+              with:
+                  name: arm_crosscompiled_samples
+                  path: scripts/tests/avh/out
+
+            - name: Install Python dependencies
+              run: |
+                  pip3 install -r scripts/tests/avh/requirements.txt
```

Comment:
consider creating a virtualev since system pip installs will stop working in newer os releases.

Reply:
Updated to use virtualenv